File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/SlugStory.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

/**
 * An object instance representing SlugStory
 *
 * @author Imran imran@quintype.com
 * @author Madhu madhu@quintype.com
 */
public class SlugStory implements Parcelable {

    @SerializedName("story")
    private Story story;

    @Override
    public String toString() {
        return "SlugStory{" +
                "story=" + story +
                '}';
    }

    /**
     * @return Slug {@link Story}
     */
    public Story story() {
        return story;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(this.story, 0);
    }

    public SlugStory() {
    }

    protected SlugStory(Parcel in) {
        this.story = in.readParcelable(Story.class.getClassLoader());
    }

    public static final Creator<SlugStory> CREATOR = new Creator<SlugStory>() {
        public SlugStory createFromParcel(Parcel source) {
            return new SlugStory(source);
        }

        public SlugStory[] newArray(int size) {
            return new SlugStory[size];
        }
    };
}

File: androidcore/src/main/java/com/example/androidcore/models/story/SlugStory.kt
package com.example.androidcore.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class SlugStory : Parcelable {
    override fun describeContents(): Int {
        return 0
    }

    @SerializedName("story")
    public val story: Story

    override fun toString(): String {
        return "SlugStory{" +
                "story=" + story +
                '}'.toString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(this.story, 0)
    }

    protected constructor(parcel: Parcel) {
        this.story = parcel.readParcelable(Story::class.java.classLoader)
    }

    companion object CREATOR : Parcelable.Creator<SlugStory> {
        override fun createFromParcel(source: Parcel): SlugStory {
            return SlugStory(source)
        }

        override fun newArray(size: Int): Array<SlugStory?> {
            return arrayOfNulls(size)
        }
    }
}
